### DISCUSSION TITLE
Propeller Force/Torque Analysis Code

### DISCUSSION THREAD

**Discussion by inse0918** [byuflowlab/FLOWUnsteady#163] [Show and tell]
The way to extract the force/torque of the propeller in 3-axis.
In this example, the number of the blade was used 2.

**Validated with .csv file**
![image](https://github.com/byuflowlab/FLOWUnsteady/assets/93040154/8061e5c3-f2bf-4da5-ba09-0bdcb5e7cf85)
- [x] tilt angle: 0 deg. 
![image](https://github.com/byuflowlab/FLOWUnsteady/assets/93040154/29b1d9d3-e47c-4870-8989-0c02fe29278f)
- [x] tilt angle: 90 deg.
- [ ] tilt angle: 45 deg.
- [ ] tilt angle: real-time var. deg.
- [ ] multiple propeller

**1. Extracting the Data**
1. B1_data: X, Y, Z forces of the blade number 1
2. B2_data: X, Y, Z forces of the blade number 2
3. Blade_Geom: RPM, Dia and Rho for calculating the coefficient of the force/torque
4. Blade_lengths: blade span for calculating the force/torque
5. Blade_radius: radius station(real R) for calculating the torque

**2. Main Matlab Code**
[main.txt](https://github.com/byuflowlab/FLOWUnsteady/files/15008599/main.txt)

**3. Modified FLOWVLM_rotor.jl**
[FLOWVLM_rotor.txt](https://github.com/byuflowlab/FLOWUnsteady/files/14990213/FLOWVLM_rotor.txt)

**Comment by inse0918** [comment]
Hi all.

Is there any way to calculate the moment of the X, Y, Z axis ?

I'm not sure about the calculation method in my Matlab code.

The forces looks well but the moment is not.

Thank you.
Best regards,
Inseo.

**Reply by ilm71** [reply]
Hi Inseo, 

I have implemented your suggested code for extracting the forces on the blade ... this works quite well - many thanks for that in advance! In terms of post processing the data I am currently investigating how the e.g. B1_data was exported and how this can be processed further, also linked to you matlab script. So you are saying, that your forces resulting from the Matlab-script are okay, but you have doubts regarding the moment? How big is the offset?

**Reply by inse0918** [reply]
Hi philm71(I don't know what can I call you sry...).

First, the thrust was quite well in all situation even the rotor is tilting in real-time.

But, the moment was not.

For example, when the tilt angle is 0 deg, and the thrust axis is X, the moment is looks good.

But, as you can see in above figure, when the tilt angle is 90 deg, the first and second element of the moment is looks weird.

When I analysis this cause, the vector of the X of the blade_2 is same with blade_1 at that first and second step.

You can check this result when you tilt the propeller with 90 deg.

Still on going, but I can not confident my code.

Thank you.
Best regards,
Inseo.

**Reply by ilm71** [reply]
Hi Inseo, 

okay understood - I am currently investigating my results, I will get back to you if I have more insights. Thank you very much again for providing your code snippet!

**Comment by ilm71** [comment]
Hi @inse0918 

I have tested and validated the outcome of my rotor in hover and the Forces and Moments seems really good so far, only the Pitching-Moment is higher than expected. 

One question for you or maybe for @EdoAlvarezR: Is it possible to extract the forces as you (@inse0918) did for several rotors like 2 distributed props? I have tried it but I was not able to access the data separately so far! 

Thanks in advance!

**Reply by ilm71** [reply]
Background of my question is that I want to investigate in detail e.g. the bending moment of the blades to experimental data for different rotor distributions and flight states. Therefore I would be pleased if you could give me a hint on how to extract this for several Rotors @EdoAlvarezR . Thanks!
